compiler/AST/primitive.cpp

```diff
@@ -702,11 +702,12 @@ initPrimitive() {
   prim_def(PRIM_REQUIRE, "require", returnInfoVoid, false, false);
 }
 
-Map<const char*, VarSymbol*> memDescsMap;
+static Map<const char*, VarSymbol*> memDescsMap;
```

Comment:
I'd like the interface for the rest of the compiler to take in Type* or const char *.
You can write a do_newMemDescInternal or whatever you choose to name it that handles the details to get both to be a Symbol*. I'd rather get the interface right than reduce the number of hashtables from 2 to 1.